disco_aws_automation/disco_datapipeline.py

```diff
@@ -0,0 +1,300 @@
+            descriptions.extend([desc for desc in batch['pipelineDescriptionList'] if _search_matches(desc)])
+        return [
+            AsiaqDataPipeline(
+                pipeline_id=meta['pipelineId'], name=meta['name'], description=meta.get('description'),
+                tags=meta.get('tags'), metadata=meta.get('fields'))
+            for meta in descriptions
+        ]
+
+    def start(self, pipeline, params=None):
+        "Activate the pipeline in AWS."
+        if not pipeline.is_persisted():
+            raise Exception("Pipeline must be saved before it can be activated")
+        param_values = _optional_dict_to_list(params) or pipeline._param_values
+        return self._dp_client.activate_pipeline(pipelineId=pipeline._id, startTimestamp=datetime.utcnow(),
+                                                 parameterValues=param_values)
+
+    def stop(self, pipeline):
+        "Deactivate the pipeline in AWS."
+        if not pipeline.is_persisted():
+            raise Exception("Pipeline must be saved before it can be deactivated")
+        return self._dp_client.deactivate_pipeline(pipelineId=pipeline._id)
+
+    def delete(self, pipeline):
+        "Remove the pipeline completely from AWS."
+        if not pipeline.is_persisted():
+            raise Exception("Pipeline must be saved before it can be deleted (though honestly...)")
+        self._dp_client.delete_pipeline(pipelineId=pipeline._id)
+
+    def fetch_or_create(self, template_name, pipeline_name, pipeline_description, tags, log_location):
+        """
+        If a pipeline with the given tags exists, return it; if not, create one with the given tags
+        and name based on the provided pipeline, save it, and return it.
+        """
+        searched = self.search_descriptions(tags=tags)
+        if searched:
+            if len(searched) > 1:
+                raise Exception("Expected one pipeline with tags %s, found %s" % (tags, len(searched)))
+            pipeline = searched[0]
+            _LOG.info("Found existing pipeline %s", pipeline._id)
+            self.fetch_content(pipeline)
+        else:
+            pipeline = AsiaqDataPipeline.from_template(
+                template_name, pipeline_name, description=pipeline_description,
+                log_location=log_location, tags=tags)
+            self.save(pipeline)
+            _LOG.info("Created new pipeline %s", pipeline._id)
+        return pipeline
+
+    def _fetch_ids(self):
+        found_defs = []
+        resp = throttled_call(self._dp_client.list_pipelines)
+        found_defs.extend(resp['pipelineIdList'])
+        while resp['hasMoreResults']:
+            resp = throttled_call(self._dp_client.list_pipelines, marker=resp['marker'])
+            found_defs.extend(resp['pipelineIdList'])
+        return found_defs
+
+
+def template_to_boto(template_json):
+    "Transform data in the format that Amazon gives in their templates to the format boto3 uses."
```

Comment:
It is—my judgement so far has been that it would be somewhat though not hugely easy and somewhat though not hugely useful (it's a transformation between two not-especially-well-specified object schemas, so...).  I can add a bit, though.